core/src/main/java/org/elasticsearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -111,7 +116,10 @@ public AllocationService(Settings settings, AllocationDeciders allocationDecider
         }
         shardsAllocators.applyFailedShards(allocation);
         reroute(allocation);
-        return new RoutingAllocation.Result(true, new RoutingTable.Builder().updateNodes(routingNodes).build().validateRaiseException(clusterState.metaData()));
+        RoutingAllocation.Result result = new RoutingAllocation.Result(true,
+                new RoutingTable.Builder().updateNodes(routingNodes).build().validateRaiseException(clusterState.metaData()));
+        logClusterHealthStateChange(clusterState, ClusterState.builder(clusterState).routingResult(result).build(), "applying failing shards");
```

Comment:
same here - would be good to have a brief summary of the shard that failed and why.